daemon/containerd/image.go

```diff
@@ -40,6 +40,7 @@ func (i *ImageService) GetImage(ctx context.Context, refOrID string, options ima
 
 	platform := platforms.AllPlatformsWithPreference(cplatforms.Default())
 	if options.Platform != nil {
+		log.G(ctx).WithField("platform", cplatforms.Format(*options.Platform)).Debug("filtering by platform")
 		platform = cplatforms.OnlyStrict(*options.Platform)
```

Comment:
It looks like we're only using the matcher for _sorting_ the results, not filtering / matching?